### DISCUSSION TITLE
Does vsomeip support for offerring multi-version of  one service on the server side?

### DISCUSSION THREAD

**Discussion by lvmy3** [COVESA/vsomeip#714] [Q&A] [answered]
Hi, there. Recently, when I read the vsomeip source code, it looks like when offerring service, the routing manager can only offer_service for a single version of one service instance. I am wondering in situation where a client and a server running different version of the same service, how they communicate?  Different minor versions of a service on client/server seems ok, right? But what about different major version? In this case, the only way they can communicate is that the server offerring multiple major version of the same service at the same time right?

**Comment by duartenfonseca** [comment, marked as answer]
hi @lvmy3 this is not possible at the moment, but is planned to be added as a functionality in the future
